smart-contracts/osmosis/contracts/cl-vault/src/vault/deposit.rs

```diff
@@ -1,42 +1,74 @@
-use cosmwasm_std::{coin, DepsMut, Env, MessageInfo, Response, Uint128, Uint256};
+use cosmwasm_std::{
+    attr, coin, Addr, Coin, Decimal, DepsMut, Env, Fraction, MessageInfo, Response, SubMsg,
+    SubMsgResult, Uint128, Uint256,
+};
 
+use osmosis_std::types::osmosis::poolmanager::v1beta1::MsgSwapExactAmountInResponse;
 use osmosis_std::types::osmosis::tokenfactory::v1beta1::MsgMint;
 
 use crate::helpers::assert::must_pay_one_or_two;
-use crate::helpers::getters::{get_asset0_value, get_depositable_tokens};
-use crate::helpers::msgs::refund_bank_msg;
+use crate::helpers::getters::{
+    get_asset0_value, get_depositable_tokens, get_swap_amount_and_direction, get_twap_price,
+};
+use crate::helpers::msgs::{refund_bank_msg, swap_msg};
+use crate::msg::DepositType;
 use crate::query::query_total_vault_token_supply;
+use crate::reply::Replies;
+use crate::state::{PoolConfig, CURRENT_SWAP_ANY_DEPOSIT};
 use crate::{
     query::query_total_assets,
     state::{POOL_CONFIG, SHARES, VAULT_DENOM},
     ContractError,
 };
 
-/// Try to deposit as much user funds as we can in the current ratio of the vault and
-/// refund the rest to the caller.
-pub(crate) fn execute_exact_deposit(
+use super::concentrated_liquidity::{get_cl_pool_info, get_position};
+use super::range::SwapDirection;
+use super::swap::{SwapCalculationResult, SwapParams};
+
+pub fn execute_deposit(
     mut deps: DepsMut,
     env: Env,
     info: MessageInfo,
     recipient: Option<String>,
+    deposit_type: DepositType,
 ) -> Result<Response, ContractError> {
-    // Unwrap recipient or use caller's address
     let recipient = recipient.map_or(Ok(info.sender.clone()), |x| deps.api.addr_validate(&x))?;
 
-    let pool = POOL_CONFIG.load(deps.storage)?;
-    let (token0, token1) = must_pay_one_or_two(&info, (pool.token0.clone(), pool.token1.clone()))?;
+    let pool_config = POOL_CONFIG.load(deps.storage)?;
+    let (token0, token1) = must_pay_one_or_two(
+        &info,
+        (pool_config.token0.clone(), pool_config.token1.clone()),
+    )?;
 
-    // get the amount of funds we can deposit from this ratio
     let (deposit, refund): ((Uint128, Uint128), (Uint128, Uint128)) =
         get_depositable_tokens(deps.branch(), token0.clone(), token1.clone())?;
 
-    let vault_denom = VAULT_DENOM.load(deps.storage)?;
-    let total_vault_shares: Uint256 = query_total_vault_token_supply(deps.as_ref())?.total.into();
-
-    let user_value = get_asset0_value(deps.storage, &deps.querier, deposit.0, deposit.1)?;
-    let refund_value = get_asset0_value(deps.storage, &deps.querier, refund.0, refund.1)?;
+    // NOTE: both the deposit arms here have some repeated math logic,
```

Comment:
If we combine both entrypoints into a single function call, why do we still have math logic that is repeated?